protocols/gossipsub/src/behaviour.rs

```diff
@@ -114,10 +114,13 @@ impl MessageAuthenticity {
     }
 }
 
+// mxinden: Wouldn't [`GossipsubEvent`] suffice? How is this /more/ generic than other structs with
+// generic type parameters?
+//
```

Comment:
Thanks for clarifying. I am in favor of being generic over the data container used here. Instead of the `Generic` prefix, why not use a default generic type parameter:

```rust
enum Event<T: AsRef<[u8]> = Vec<u8>> {
    Message(T),
}
```

I did not grasp the connection between additional compression and the `FastMessageId` with the code and comments only. Would be great if you could either as suggested rename `FastMessageId` or add additional comments describing your use case as an example.